integrations/qdrant/src/haystack_integrations/document_stores/qdrant/filters.py

```diff
@@ -141,34 +133,47 @@ def _build_nin_condition(self, key: str, value: List[models.ValueVariants]) -> m
             must_not=[
                 (
                     models.FieldCondition(key=key, match=models.MatchText(text=item))
-                    if isinstance(item, str) and " " not in item
+                    if isinstance(item, str) and " " in item
                     else models.FieldCondition(key=key, match=models.MatchValue(value=item))
                 )
                 for item in value
             ]
         )
 
-    def _build_lt_condition(self, key: str, value: float) -> models.Condition:
+    def _build_lt_condition(self, key: str, value: Union[str, float, int]) -> models.Condition:
+        if isinstance(value, str) and is_datetime_string(value):
+            return models.FieldCondition(key=key, range=models.DatetimeRange(lt=value))
+
         if not isinstance(value, (int, float)):
-            msg = f"Value {value} is not an int or float"
+            msg = f"Value {value} is not an int or float or datetime string"
             raise FilterError(msg)
```

Comment:
I would do something like this:
```python
    msg = f"Value {value} is not an int or float or datetime string"
    if isinstance(value, str):
        if is_datetime_string(value):
            return models.FieldCondition(key=key, range=models.DatetimeRange(lt=value))
       raise FilterError(msg)

        if not isinstance(value, (int, float)):
          raise FilterError(msg)
```

It is important that if the string is not recognized as a datetime object, an error is raised.

Reply:
Would https://github.com/deepset-ai/haystack-core-integrations/pull/570/commits/dd821d05269183a899b3b97cd88828790f9931e5 do?